codec_record.go

```diff
@@ -114,8 +114,8 @@ func (d *structDecoder) Decode(ptr unsafe.Pointer, r *Reader) {
 			}
 
 			if f.Type().Kind() == reflect.Ptr {
-				if *((*unsafe.Pointer)(ptr)) == nil {
-					newPtr := f.Type().UnsafeNew()
+				if *((*unsafe.Pointer)(fieldPtr)) == nil {
+					newPtr := reflect2.Type2(f.Type().Type1().Elem()).UnsafeNew()
```

Comment:
```suggestion
					newPtr := f.Type().(*reflect2.UnsafePtrType).Elem().UnsafeNew()
```
Rather than jumping into Type1, it would better to cast to un unsafe ptr type to get the Elem.

Reply:
Ah, thank you, I am unfamiliar with `reflect2` and couldn't see how to do it :+1:.